source/common/network/tcp_listener_impl.cc

```diff
@@ -118,5 +124,9 @@ void TcpListenerImpl::enable() { file_event_->setEnabled(Event::FileReadyType::R
 
 void TcpListenerImpl::disable() { file_event_->setEnabled(0); }
 
+void TcpListenerImpl::setRejectFraction(const float reject_fraction) {
+  reject_fraction_ = reject_fraction;
```

Comment:
Can you ASSERT 0 <= reject_fraction <= 1? I think the Bernoulli function correctly handles out of bounds but might as well try to check for obvious config/code bugs. (Bonus points would be to introduce a struct wrapper for the float which would check this at point of assignment, etc. but up to you.)